Neo4j.Driver/Neo4j.Driver/Internal/RetryLogic.cs

```diff
@@ -66,11 +69,12 @@ public T Retry<T>(Func<T> runTxFunc)
                 {
                     return runTxFunc();
                 }
-                catch (Exception e) when (e is SessionExpiredException || e is TransientException || e is ServiceUnavailableException)
+                catch (Exception e) when (e is SessionExpiredException || e.IsRecoverableError() || e is ServiceUnavailableException)
```

Comment:
Should newly introduced `#IsRetriableTransientError()` be used here instead of `#IsRecoverableError()`?
Maybe also add tests asserting that retries do not happen on `Terminated` and `LockClientStopped`.